Assets/Scripts/Models/HealthSystem.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+
+[Serializable]
+public class HealthSystem
+{
+    //all of this data should probably be generated at runtime or loaded from a file
+    public float Value
+    {
+        get
+        {
+            return _currentHealth;
+        }
+
+        private set
+        {
+            if (Invincible == true) return;
+            //We have been Revived!
+            if (_currentHealth == 0 && value > _currentHealth)
+            {
+                _currentHealth = value;
+
+                if (OnRevive != null)
+                    OnRevive();
+                return;
+            }
+            //we have been hit but we are not dead!
+            if (value < _currentHealth && _currentHealth > 0)
```

Comment:
@MANDAL0R3 You are 100% free to use my change. All my contributions are MIT/Apache 2.0 multi-licensed if that's your question.

And copyright doesn't really cover 1 line ;) At least in Switzerland where I am.